src/main/java/lv/ctco/javaschool/goal/boundary/GoalApi.java

```diff
@@ -1,31 +1,146 @@
+import lv.ctco.javaschool.goal.entity.GoalDto;
+import lv.ctco.javaschool.goal.entity.GoalFormDto;
+import lv.ctco.javaschool.goal.entity.Tag;
 
 import javax.annotation.security.RolesAllowed;
-import javax.inject.Inject;
-import javax.persistence.EntityManager;
-import javax.persistence.PersistenceContext;
 import javax.ejb.Stateless;
-import javax.ws.rs.Path;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
 import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
+import static java.time.temporal.ChronoUnit.DAYS;
 
 @Path("/goal")
 @Stateless
 public class GoalApi {
-    @PersistenceContext
-    private EntityManager em;
     @Inject
     private UserStore userStore;
     @Inject
     private GoalStore goalStore;
 
+    @GET
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/mygoals")
+    public List<GoalDto> getMyGoals() {
+        User currentUser = userStore.getCurrentUser();
+        List<Goal> goalsList = goalStore.getGoalsListFor(currentUser);
+        if (goalsList.size() != 0) {
+            return goalsList.stream()
+                    .map(this::convertToDto)
+                    .collect(Collectors.toList());
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    @GET
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/mygoals/{id}")
+    public GoalDto getGoalDtoByGoalId(@PathParam("id") long goalId) {
+        Optional<Goal> goal = goalStore.getGoalById(goalId);
+        if (goal.isPresent()) {
+            Goal g = goal.get();
+            return convertToDto(g);
+        } else {
+            return new GoalDto();
+        }
+    }
+
     @POST
     @RolesAllowed({"ADMIN", "USER"})
-    public void startPage() {
-        /// place for methods on page reload
-        /// currently - none
+    @Path("/newgoal")
+    public void createNewGoal(GoalFormDto goalDto) {
+        User user = userStore.getCurrentUser();
+        Goal goal = new Goal();
+        if (!goalDto.getGoalMessage().isEmpty() && !goalDto.getDeadline().isEmpty()) {
+            goal.setGoalMessage(goalDto.getGoalMessage());
+            goal.setTags(parseStringToTags(goalDto.getGoalMessage()));
+
+            DateTimeFormatter formatterD = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+            LocalDate localDate = LocalDate.parse(goalDto.getDeadline(), formatterD);
+            goal.setDeadlineDate(localDate);
+
+            goal.setUser(user);
+            goal.setRegisteredDate(LocalDateTime.now());
+            goalStore.addGoal(goal);
+        }
+    }
+
+    GoalDto convertToDto(Goal goal) {
```

Comment:
As mentioned above - please make a dedicated class for DTO conversion and extract such functions into that class.

Reply:
Done